src/opm/output/eclipse/CreateInteHead.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+  Copyright (c) 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/output/eclipse/WriteRestartHelpers.hpp>
+
+#include <opm/output/eclipse/InteHEAD.hpp>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Tuning.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/Regdims.hpp>
+#include <opm/parser/eclipse/EclipseState/Tables/TableManager.hpp>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <exception>
+#include <stdexcept>
+#include <vector>
+
+namespace {
+    Opm::RestartIO::InteHEAD::WellTableDim
+    getWellTableDims(const ::Opm::Runspec&  rspec,
+                     const ::Opm::Schedule& sched,
+                     const std::size_t      step)
```

Comment:
Yes - there are bugs with these time steps, before and after and all that. I would strongly prefer if we go in deep n..i mode and:

1. Enforce that all time step variables have a prefix - i.e. `report_step` for the current. 
2. Instead of doing: `std::size_t step_minus_one = step == 0 ? 0 : step - 1;` I would prefer if the functions (when needed) get an additional argument: `sim_step` (which is typically `report_step - 1`) - that makes it clearer.

Then well queries e.t.c will typically be done with the `sim_step` argument.